frontend/src/components/PlanForm.tsx

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import React, { useState } from 'react'
+import { SearchBarFilter } from './SearchBarFilter'
+import './PlanForm.css'
+import axios, { AxiosError } from 'axios'
+
+enum EventType {
+  BIRTHDAY = 'BIRTHDAY',
+  ANNIVERSARY = 'ANNIVERSARY',
+  GRADUATION = 'GRADUATION',
+  WEDDING = 'WEDDING',
+  BABY_SHOWER = 'BABY_SHOWER',
+  ACHIEVEMENT = 'ACHIEVEMENT',
+  HOLIDAY = 'HOLIDAY',
+  OTHER = 'OTHER'
+}
+
+export interface Friend {
+  firstName: string
+}
+
+export const PlanForm = (): JSX.Element => {
+  const [specialPerson, setSpecialPerson] = useState<Friend | undefined>()
+  const [description, setDescription] = useState<string>('')
+  const [startDate, setStartDate] = useState<string>('')
+  const [endDate, setEndDate] = useState<string>('')
+  const [specialDate, setSpecialDate] = useState<string>('')
+  const [friends, setFriends] = useState<Friend[]>([])
+  const [errorMessage, setErrorMessage] = useState<string>('')
+  const [error, setError] = useState<boolean>(false)
+  const [maxFriends] = useState<number>(3)
+  const [eventType, setEventType] = useState<EventType>()
+
+  const handleSpecialPersonChange = (friend: Friend): void => {
+    setSpecialPerson(friend)
+  }
+
+  const handleSpecialPersonRemove = (friend: Friend[] | undefined): void => {
+    friend !== undefined && setSpecialPerson(friend[0])
+  }
+
+  const handleDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>): void => {
+    setDescription(event.target.value)
+  }
+
+  const handleStartDateChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setStartDate(event.target.value)
+  }
+
+  const handleEndDateChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setEndDate(event.target.value)
+  }
+
+  const handleSpecialDateChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setSpecialDate(event.target.value)
+  }
+
+  const handleFriendsChange = (friend: Friend): void => {
+    if (friends.length >= maxFriends) {
+      handleError(`Only a max of ${maxFriends} friends are allowed`)
+    } else if (!friends.includes(friend)) {
+      setFriends(friends => [...friends, friend])
+    }
+  }
+
+  const hanldeRemoveFriends = (friend: Friend[]): void => {
+    setFriends(friend)
+  }
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>): Promise<void> => {
+    event.preventDefault()
+    if (specialPerson === undefined) {
+      handleError('Please add 1 special person')
+      return
+    }
+    try {
+      await axios.post('http://localhost:8000/planform', {
+        specialPerson,
+        description,
+        startDate,
+        endDate,
+        specialDate,
+        friends,
+        eventType
+      }, {
+        withCredentials: true
+      })
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        console.error(error?.response?.status)
+        console.error(error?.response?.data)
+      }
```

Comment:
anything else you want to do if the post request doesn't work?

Reply:
Ohhh display an error message on UI on why the submission didn't work.

Reply:
yeaup!